controller/tests/array_action/svc/array_mediator_svc_test.py

```diff
@@ -182,6 +187,25 @@ def test_delete_volume_return_volume_delete_errors(self, mock_warning):
         with self.assertRaises(CLIFailureError):
             self.svc.delete_volume("vol")
 
+    @patch("controller.array_action.array_mediator_svc.is_warning_message")
+    def test_delete_volume_have_snapshot_fcmaps_not_removed(self, mock_warning):
+        mock_warning.return_value = False
+        fcmaps = self.fcmaps.copy()
+        fcmaps[0].copy_rate = "0"
+        self.fcmaps.clear()
+        self.svc.client.svcinfo.lsfcmap.side_effect = [Mock(as_list=[]), Mock(as_list=fcmaps)]
+        with self.assertRaises(array_errors.ObjectIsStillInUseError):
+            self.svc.delete_volume("vol")
+
+    @patch("controller.array_action.array_mediator_svc.is_warning_message")
+    def test_delete_volume_have_clone_fcmaps_removed(self, mock_warning):
```

Comment:
do we have a UT for fcmaps in both directions (as_source + as_target)?

Reply:
yes, see test_delete_snapshot_success